go/libraries/doltcore/schema/alterschema/dropcolumn.go

```diff
@@ -71,15 +76,60 @@ func DropColumn(ctx context.Context, tbl *doltdb.Table, colName string) (*doltdb
 
 	rd, err := tbl.GetRowData(ctx)
 
+	prunedRowData, err := dropColumnValuesForTag(ctx, tbl.Format(), newSch, rd, dropTag)
+
 	if err != nil {
 		return nil, err
 	}
 
-	newTable, err := doltdb.NewTable(ctx, vrw, schemaVal, rd)
+	newTable, err := doltdb.NewTable(ctx, vrw, schemaVal, prunedRowData)
 
 	if err != nil {
 		return nil, err
 	}
 
 	return newTable, nil
 }
+
+func dropColumnValuesForTag(ctx context.Context, nbf *types.NomsBinFormat, newSch schema.Schema, rowData types.Map, dropTag uint64) (types.Map, error) {
+	re := rowData.Edit()
+
+	mi, err := rowData.BufferedIterator(ctx)
+
+	if err != nil {
+		return types.EmptyMap, err
+	}
+
+	for {
+		k, v, err := mi.Next(ctx)
+
+		if k == nil || v == nil {
```

Comment:
Doesn't this iterator return err == io.EOF when it has no more elements? More common in go to match this error than check a nil value returned by the iterator.

Reply:
it doesn't return io.EOF the types package just gives `nil`